diff --git a/nova/api/ec2/ec2utils.py b/nova/api/ec2/ec2utils.py
index b49b9ff..17cdee3 100644
--- a/nova/api/ec2/ec2utils.py
+++ b/nova/api/ec2/ec2utils.py
@@ -42,6 +42,7 @@ def memoize(func):
         if not _CACHE:
             _CACHE = memorycache.get_client()
         key = "%s:%s" % (func.__name__, reqid)
+        key = str(key)
         value = _CACHE.get(key)
         if value is None:
             value = func(context, reqid)
